File: LeetCode/71SimplifyPath/SimplifyPath.cpp
class Solution {
public:
    string simplifyPath(string path) {
        // Stack -> Time:O(n), Space:O(n)
        stack<string> s;
        path = path + '/';
        string curr = "";
        string ans = "";
        for (int i  = 0; i < path.size(); i++) {
            if (path[i] == '/') {
                if (curr == "..") {
                    if (!s.empty()) {
                        s.pop();
                    }
                }
                else if (curr != "." && curr != "") {
                    s.push(curr);
                }
                curr = "";
            }
            else {
                curr += path[i];
            }
        }
        while (!s.empty()) {
            if (ans == "") {
                ans += s.top();
            }
            else {
                ans = s.top() + '/' + ans;
            }
            s.pop();
        }
        ans = '/' + ans;
        return ans;
    }
};

File: LeetCode/71SimplifyPath/SimplifyPath.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        # Stack -> Time:O(n), Space:O(n)
        s = []
        curr = ""
        for i in path + "/":
            if i == '/':
                if curr == "..":
                    if s != []:
                        s.pop()
                elif curr != "" and curr != '.':
                    s.append(curr)
                curr = ""
            else:
                curr += i
        return '/' + '/'.join(s)
